src/adminEntity.ts

```diff
@@ -91,6 +90,38 @@ abstract class AdminEntity {
     validateFieldsExist(this, 'searchFields', this.metadata)
     validateFieldsAreNotRelation(this, 'searchFields', this.metadata)
   }
+
+  buildSearchQueryOptions(
+    query: SelectQueryBuilder<unknown>,
+    options: { alias: string; searchParam: string },
+  ) {
+    if (options.searchParam && this.searchFields) {
+      const searchArray = options.searchParam.split(' ')
+      searchArray.forEach((searchTerm, searchTermIndex) =>
+        query.andWhere(
+          new Brackets((qb: SelectQueryBuilder<unknown>) => {
+            this.searchFields.forEach((field, fieldIndex) => {
+              const paramString = `searchTerm${searchTermIndex}Field${fieldIndex}`
+              qb.orWhere(`${options.alias}.${field} LIKE :${paramString}`, {
+                [paramString]: `%${searchTerm}%`,
+              })
+            })
+            return qb
+          }),
+        ),
+      )
+    }
+    return query
+  }
+
+  buildPaginationQueryOptions(query: SelectQueryBuilder<unknown>, page: number) {
+    query.skip(this.resultsPerPage * (page - 1)).take(this.resultsPerPage)
+    return query
+  }
+
+  getManyAndCount(query: SelectQueryBuilder<unknown>) {
+    return query.getManyAndCount()
+  }
```

Comment:
That doesn't look like it belongs here? What abstraction are you trying to provide here?